compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1103,10 +1109,6 @@ pub enum Rvalue<'tcx> {
 
     /// Same as `BinaryOp`, but yields `(T, bool)` with a `bool` indicating an error condition.
     ///
-    /// When overflow checking is disabled and we are generating run-time code, the error condition
-    /// is false. Otherwise, and always during CTFE, the error condition is determined as described
-    /// below.
-    ///
     /// For addition, subtraction, and multiplication on integers the error condition is set when
     /// the infinite precision result would be unequal to the actual result.
     ///
```

Comment:
The code generation doesn't implement this behavior: https://github.com/rust-lang/rust/blob/5b450244876154bc1bd134694398e80c12e00b5c/compiler/rustc_codegen_ssa/src/mir/rvalue.rs#L658-L661

Reply:
I'm wondering what to do with this. There are 2 options:
1. keep this impl as an "optimization";
2. consider that LLVM is smart enough and remove that edge case.

For now, I added a comment stating (1), but I'm ready to switch to (2) for consistency.

Reply:
The second option is mostly an empirical question of compile time impact. The opt-level=1 looks sufficient to optimize out overflow intrinsics.

Adjusting the spec or code generation either way seems fine to me, as long as code generation correctly implements the spec.

Reply:
Users care about debug codegen: https://github.com/rust-lang/rust/issues/72751

We should take some care for debug-mode codegen, because that also improves the experience of using a debugger. And some users don't even use LLVM (and we should expect that to become more common).

Reply:
I was expecting that the existing CheckedBinaryOp docs would be unchanged (except for wording tweaks to not talk about code generation any more)... if this part is removed from the docs then yeah codegen and Miri should be adjusted accordingly. It is in theory possible to have MIR that uses the CheckedBinaryOp not for an assertion but for something else.

Reply:
I removed the special case in codegen. It appears that this doesn't have any effect on compile-time perf.